modules/improvedigitalBidAdapter.js

```diff
@@ -1,17 +1,456 @@
-import { deepSetValue, logError, _each, getBidRequest, isNumber, isArray, deepAccess, isFn, isPlainObject, logWarn, getBidIdParameter, getUniqueIdentifierStr, isEmpty, isInteger } from '../src/utils.js';
-import { registerBidder } from '../src/adapters/bidderFactory.js';
-import { config } from '../src/config.js';
-import { BANNER, NATIVE, VIDEO } from '../src/mediaTypes.js';
+import {
+  cleanObj,
+  deepAccess,
+  deepClone,
+  deepSetValue,
+  getAdUnitSizes,
+  getBidIdParameter,
+  getBidRequest,
+  getDNT,
+  getUniqueIdentifierStr,
+  isArray,
+  isFn,
+  isNumber,
+  isPlainObject,
+  logWarn, mergeDeep,
+  parseGPTSingleSizeArrayToRtbSize
+} from '../src/utils.js';
+import {registerBidder} from '../src/adapters/bidderFactory.js';
+import {config} from '../src/config.js';
+import {BANNER, NATIVE, VIDEO} from '../src/mediaTypes.js';
 import {Renderer} from '../src/Renderer.js';
-import { createEidsArray } from './userId/eids.js';
-import includes from 'core-js-pure/features/array/includes.js';
+import {createEidsArray} from './userId/eids.js';
 
 const BIDDER_CODE = 'improvedigital';
 const RENDERER_URL = 'https://acdn.adnxs.com/video/outstream/ANOutstreamVideo.js';
-const VIDEO_TARGETING = ['skip', 'skipmin', 'skipafter'];
+const REQUEST_URL = 'https://ad.360yield.com/pb';
+const VIDEO_PARAMS = {
+  DEFAULT_MIMES: ['video/x-ms-wmv', 'video/mp4'],
+  SUPPORTED_PROPERTIES: ['mimes', 'minduration', 'maxduration', 'protocols', 'w', 'h', 'startdelay', 'placement', 'linearity', 'skip', 'skipmin',
+    'skipafter', 'sequence', 'battr', 'maxextended', 'minbitrate', 'maxbitrate', 'boxingallowed', 'playbackmethod', 'playbackend', 'delivery', 'pos', 'companionad',
+    'api', 'companiontype', 'ext'],
+  PLACEMENT_TYPE: {
+    INSTREAM: 1,
+    OUTSTREAM: 3,
+  }
+};
+const NATIVE_DATA = {
+  ASSET_TYPES: {
+    TITLE: 'title',
+    IMG: 'img',
+    DATA: 'data',
+  },
+  PARAMS: {
+    title: {id: 0, name: 'title', assetType: 'title', default: {len: 50}},
+    sponsoredBy: {id: 1, name: 'sponsoredBy', assetType: 'data', type: 1},
+    icon: {id: 2, name: 'icon', assetType: 'img', type: 2},
+    body: {id: 3, name: 'body', assetType: 'data', type: 2},
+    image: {id: 4, name: 'image', assetType: 'img', type: 3},
+    rating: {id: 5, name: 'rating', assetType: 'data', type: 3},
+    likes: {id: 6, name: 'likes', assetType: 'data', type: 4},
+    downloads: {id: 7, name: 'downloads', assetType: 'data', type: 5},
+    price: {id: 8, name: 'price', assetType: 'data', type: 6},
+    salePrice: {id: 9, name: 'salePrice', assetType: 'data', type: 7},
+    phone: {id: 10, name: 'phone', assetType: 'data', type: 8},
+    address: {id: 11, name: 'address', assetType: 'data', type: 9},
+    body2: {id: 12, name: 'body2', assetType: 'data', type: 10},
+    displayUrl: {id: 13, name: 'displayUrl', assetType: 'data', type: 11},
+    cta: {id: 14, name: 'cta', assetType: 'data', type: 12},
+  },
+  DEFAULT_REQUEST: {
```

Comment:
What's the reasoning for this? IMHO it's not a good idea if publisher forgets to define assets to just respond with title. I can't imagine a good-looking creative with title only.

Reply:
@jbartek25, Actually the minimum requirement for a default native request is to send the title, otherwise, the validation fails. I kept it for a validation purpose, so mistakenly if the publisher does not include the title we will add it. Should we include a couple of more information e.g: body, image, sponsoredBy?

Reply:
> Actually the minimum requirement for a default native request is to send the title, otherwise, the validation fails

What validation? You can request just body and it works just fine. Please remove the default completely. We are not going to second-guess publisher's intentions.

Reply:
@jbartek25 sorry, my bad luck. I misunderstood, removing the default.